experiments/AMIP/moist_mpi_earth/coupler_utils/bcfile_reader.jl

```diff
@@ -192,3 +197,40 @@ function update_midmonth_data!(date, bcf_info)
 end
 
 next_date_in_file(bcfile_info) = bcfile_info.all_dates[bcfile_info.segment_idx[1] + Int(1)]
+
+# IO - daily
+"""
+    interpolate_midmonth_to_daily(date, bcf_info)
+
+Interpolates linearly between two `Fields` in the `bcf_info` struct, or returns the first Field if interpolation is switched off. 
+"""
+function interpolate_midmonth_to_daily(date, bcf_info)
+    @show bcf_info.segment_length[1]
+    if bcf_info.interpolate_daily[1] && bcf_info.segment_length[1] > FT(0)
+        segment_length = bcf_info.segment_length
+        segment_idx = bcf_info.segment_idx
+        all_dates = bcf_info.all_dates
+        monthly_fields = bcf_info.monthly_fields
+
+        return intepol.(
+            monthly_fields[1],
+            monthly_fields[2],
+            (date - all_dates[Int(segment_idx[1])]).value,
+            segment_length[1],
+            FT,
+        )
+    else
+        return bcf_info.monthly_fields[1]
+    end
+end
+
+"""
+    intepol(f_t1, f_t2, t, Δt, FT)
+
+Performs linear interpolation within a segment `Δt` = (t1 - t2), of fields `f_t1` and `f_t2`, with `t`(`f_t1`) = t1.
+"""
+function intepol(f1, f2, t, Δt, FT)
+    interp_fraction = FT(t) / Δt
+    @assert abs(interp_fraction) <= FT(1) "time interpolation weights must be <= 1, but `interp_fraction` = $interp_fraction"
```

Comment:
I dont think this formula is correct. 
For example:
If t = t1, we get f1*t1/Delta +f2*(Delta-t1)/Delta = 
f1*t1/Delta - f2 *t2/Delta. It should be f1.

I think it should be f = (f2-f1)/(t2-t1)(t-t1) + f1. Then f(t1) = f1, f(t2) = f2, and f((t1+t2)/2) = (f1+f2)/2.

(Also, it should not matter if t is negative, the formula should work still.)

Reply:
Ahh, the notation is a little confusing. 🤯  `t` here actually does refer to difference between the current date and `t1`, which I believe makes it equivalent to your suggestion. I failed at trying to be less verbose, haha! I will call it `\Delta_tt1` and `\Delta_t1t2`.  This is why reviews are super helpful! :) As for the assert, this is just catching cases where the dates are not passed in properly, so the current date is outside of range [t1,t2].